LoadException: Root value already specified

So far, each `FXMLView` injects the same `FXMLRootProvider` object, which is annotated as Singleton, so it uses the same `FXMLLoader` object.

When there are more than one FXML file, this causes:

```
Caused by: javafx.fxml.LoadException: Root value already specified.
        at javafx.fxml/javafx.fxml.FXMLLoader.constructLoadException(FXMLLoader.java:2703)
        at javafx.fxml/javafx.fxml.FXMLLoader.createElement(FXMLLoader.java:2856)
        at javafx.fxml/javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2805)
        at javafx.fxml/javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2634)
        at javafx.fxml/javafx.fxml.FXMLLoader.load(FXMLLoader.java:2532)
        at com.gluonhq.ignite.micronaut.FXMLRootProvider.getRootByClass(FXMLRootProvider.java:59)
 
```
which prevents the use of more than one FXMLView per application.

As a quick fix, removing the `@Singleton` annotation from `FXMLRootProvider` seems to solve the problem. 
Alternatively, `loader` can be replaced with `ctx.getBean(FXMLLoader.class)`.

--- a/micronaut/src/main/java/com/gluonhq/ignite/micronaut/FXMLRootProvider.java
+++ b/micronaut/src/main/java/com/gluonhq/ignite/micronaut/FXMLRootProvider.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, Gluon
+ * Copyright (c) 2020, 2021, Gluon
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -28,7 +28,6 @@
 package com.gluonhq.ignite.micronaut;
 
 import io.micronaut.context.ApplicationContext;
-import javafx.fxml.FXMLLoader;
 import javafx.scene.Node;
 import javafx.scene.Scene;
 import org.slf4j.Logger;
@@ -47,7 +46,7 @@ public class FXMLRootProvider {
     private static final Logger LOG = LoggerFactory.getLogger(FXMLRootProvider.class);
 
     @Inject private ApplicationContext ctx;
-    @Inject FXMLLoader loader;
+    @Inject FXMLLoaderFactory loaderFactory;
 
     public Node getRootByClass(@NotBlank Class<?> viewClass) {
         String viewPath = getViewPath(viewClass);
@@ -56,9 +55,9 @@ public class FXMLRootProvider {
 
         try {
             LOG.info("Attempting to load " + fxml);
-            node = loader.load(viewClass.getResourceAsStream(fxml));
+            node = loaderFactory.getLoader().load(viewClass.getResourceAsStream(fxml));
         } catch (IOException e) {
-            throw new RuntimeException("Could find resource " + fxml, e);
+            throw new RuntimeException("Error loading resource " + fxml, e);
         }
 
         // Make sure that CSS is added to a scene that node is going to be added to

